File: src/main/java/models/Facture.java
package models;

public class Facture {
    private int id_fact;
    private String client;
    private String caissier;
    private double montant;
    private double percu;
    private double rendu;

    public Facture() {
    }

    public Facture(int id_fact ,String client, String caissier, double montant, double percu, double rendu) {
        this.id_fact = id_fact;
        this.client = client;
        this.caissier = caissier;
        this.montant = montant;
        this.percu = percu;
        this.rendu = rendu;
    }

    public int getId() {
        return id_fact;
    }

    public void setId(int id_fact) {
        this.id_fact = id_fact;
    }

    public String getClient() {
        return client;
    }

    public void setClient(String client) {
        this.client = client;
    }

    public String getCaissier() {
        return caissier;
    }

    public void setCaissier(String caissier) {
        this.caissier = caissier;
    }

    public double getMontant() {
        return montant;
    }

    public void setMontant(double montant) {
        this.montant = montant;
    }

    public double getPercu() {
        return percu;
    }

    public void setPercu(double percu) {
        this.percu = percu;
    }

    public double getRendu() {
        return rendu;
    }

    public void setRendu(double rendu) {
        this.rendu = rendu;
    }

    @Override
    public String toString() {
        return "Facture{" +
                "id_fact=" + id_fact +
                ", client='" + client + '\'' +
                ", caissier='" + caissier + '\'' +
                ", montant=" + montant +
                ", percu=" + percu +
                ", rendu=" + rendu +
                '}';
    }
}

File: src/Entity/Facture.php
<?php

namespace App\Entity;

use App\Repository\FactureRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: FactureRepository::class)]
class Facture
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private ?int $id;

    #[ORM\Column(type: 'string', length: 255)]
    private ?string $client;

    #[ORM\Column(type: 'string', length: 255)]
    private ?string $caissier;

    #[ORM\Column(type: 'decimal', precision: 10, scale: 2)]
    private ?float $montant;

    #[ORM\Column(type: 'decimal', precision: 10, scale: 2)]
    private ?float $percu;

    #[ORM\Column(type: 'decimal', precision: 10, scale: 2)]
    private ?float $rendu;


    #[ORM\ManyToOne(targetEntity: Panier::class)]
    #[ORM\JoinColumn(name: 'id_panier', referencedColumnName: 'id')]
    private ?Panier $panier;
    
    public function getId(): ?int
    {
        return $this->id;
    }

    public function getClient(): ?string
    {
        return $this->client;
    }

    public function setClient(string $client): self
    {
        $this->client = $client;

        return $this;
    }

    public function getCaissier(): ?string
    {
        return $this->caissier;
    }

    public function setCaissier(string $caissier): self
    {
        $this->caissier = $caissier;

        return $this;
    }

    public function getMontant(): ?float
    {
        return $this->montant;
    }

    public function setMontant(float $montant): self
    {
        $this->montant = $montant;

        return $this;
    }

    public function getPercu(): ?float
    {
        return $this->percu;
    }

    public function setPercu(float $percu): self
    {
        $this->percu = $percu;

        return $this;
    }

    public function getRendu(): ?float
    {
        return $this->rendu;
    }

    public function setRendu(float $rendu): self
    {
        $this->rendu = $rendu;

        return $this;
    }



    public function getPanier(): ?Panier
    {
        return $this->panier;
    }

    public function setPanier(?Panier $panier): self
    {
        $this->panier = $panier;

        return $this;
    }
}
